src/core/server/saved_objects/routes/migrate.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { IRouter } from '../../http';
+import { IKibanaMigrator } from '../migrations';
+
+export const registerMigrateRoute = (
+  router: IRouter,
+  migratorPromise: Promise<IKibanaMigrator>
+) => {
+  router.post(
+    {
+      path: '/_migrate',
```

Comment:
If we're only using this endpoint internally for the functional test runner and don't intend for end-users in production to use this endpoint, I wouldn't tie this to the `savedObjectsManagement:all` privilege. End-users who get all access to the saved-object management feature would then be able to use this.

This route is rather different than our other "internal" routes. Our other internal routes are intended to be used from the browser, but end-users are authorized to access them. We don't really want this route to be used in production at all. We definitely can use the `access:` tag on the route to restrict which users should have access to this route. However, we really only want it to be used by the functional test runner, which I don't believe has a real user to represent. If we used an `access:migrateSavedObjects` tag here, and then didn't add it to any privileges only a `superuser` would be able to invoke this API, as they're authorized to do everything.

Reply:
The other stipulation is that relying on the security plugin's authorization won't apply when the user hasn't enabled security, which is the default. What are the repercussions of any user being able to being migrations in these scenarios?

Reply:
I'm trying to come up with a bad scenario and coming up empty. Migrations only get run on objects from older versions, which the only way you could create objects from older versions is manually via direct Elasticsearch API calls, which on superusers and kibana_system should have access to do.

Adding `access:migrateSavedObjects` tag makes sense to me as an extra layer of precaution, but I don't think we need to do anything extra for the no-security scenario. Does that work for y'all?

Reply:
Having just `access:migrateSavedObjects` sounds like a good trade-off to me assuming it works with `esArchiver` and there is no reliable way no detect "FTR" scenario using `env.packageInfo.dist/buildNum` (to not register this endpoint at all otherwise or register it via custom test plugin or something along these lines).

Reply:
> there is no reliable way no detect "FTR" scenario using env.packageInfo.dist/buildNum (to not register this endpoint at all otherwise or register it via custom test plugin or something along these lines).

Yeah we decided against having a "test-mode" of any sort because we want to avoid creating any situation where the build under test in FTR behaves differently than production.

Reply:
> Having just access:migrateSavedObjects sounds like a reasonable trade-off to me assuming it works with esArchiver

I'm not very aware of how tags works. Should I just add the tag to the route definition, or is there anything else to do to make this actually work, specially with the FTR?

Reply:
> Should I just add the tag to the route definition, or is there anything else to do to make this actually work

Yeah, technically you just need to include this one into tags of the specific route and that's it. But I just realized that we don't do API authorization checks for `/internal` routes, @kobelb was that intentional decision for some reason or we just didn't have any `/internal` routes that would require authorization check at that time.

> specially with the FTR?

I'm not sure what user context is used to call this endpoint to be honest, if it's superuser then it should just work.

Reply:
> I'm not sure what user context is used to call this endpoint to be honest, if it's superuser then it should just work.

At least when launching locally, it's the `elastic` user yes.

Reply:
> @kobelb was that intentional decision for some reason or we just didn't have any /internal routes that would require authorization check at that time.

There were no `/internal` routes when this was written, and we decided to only run that route middleware when the url began with `/api`. We can likely remove that restriction entirely, as we're only performing the authorization logic on route which have the `access:` tag. I created https://github.com/elastic/kibana/pull/58351 to see the impact this has. @legrego do you recall any reason why we included the restriction that the route must start with `/api` originally?

Reply:
@kobelb nope I don’t think we had a great reason, beyond what you already mentioned that we didn’t have any /internal routes at that time. I don’t see a problem removing this restriction

Reply:
Just pushed the commit adding the access tag to the route, to see if that doesn't introduce another 'surprise' from the FTR call. Waiting for https://github.com/elastic/kibana/pull/58351 to change the endpoint from `api` to `internal`.

Reply:
CI just passed on https://github.com/elastic/kibana/pull/58351, so I think we should be good, just waiting on a review at this point.

Reply:
I see  #58351 has been merged. Thanks a lot to the whole security team 😄 Will move the endpoint to the `internal` prefix tomorrow, and we should be good to go.